src/Files.App/Utils/Cloud/Detector/GoogleDriveCloudDetector.cs

```diff
@@ -74,13 +95,23 @@ await FilesystemTasks.Wrap(() => StorageFile.GetFileFromPathAsync(Path.Combine(a
 				if (string.IsNullOrWhiteSpace(path))
 					continue;
 
+				if (!AddMyDriveToPathAndValidate(ref path))
+				{ 
+					_logger.LogWarning("Validation failed for " + path + " (media)");
+					continue;
+				}
+
 				var folder = await StorageFolder.GetFolderFromPathAsync(path);
 				string title = reader["name"]?.ToString() ?? folder.Name;
-				string iconPath = Path.Combine(Environment.GetEnvironmentVariable("ProgramFiles"), "Google", "Drive File Stream", "drive_fs.ico");
 
 				App.AppModel.GoogleDrivePath = path;
 
-				StorageFile iconFile = await FilesystemTasks.Wrap(() => StorageFile.GetFileFromPathAsync(iconPath).AsTask());
+				StorageFile? iconFile = await GetGoogleDriveIconFileAsync();
```

Comment:
Is var more appropriate here?
```suggestion
				var iconFile = await GetGoogleDriveIconFileAsync();
```